index.bs

```diff
@@ -3701,6 +3727,792 @@ opened</dfn> steps given |window|, |type| and |message|.
+<code>network.beforeRequestSent</code> event has already been sent.
+
+### Types ### {#module-network-types}
+
+#### The network.BaseParameters type #### {#type-network-BaseParameters}
+
+<pre class="cddl local-cddl">
+network.BaseParameters = {
+    request: network.RequestData,
+    navigation: Navigation / null,
+    context: BrowsingContext / null,
+    timestamp: int,
+    isRedirect: bool,
+}
+</pre>
+
+The <code>network.BaseParameters</code> type is an abstract type representing
+the data that's common to all network events.
+
+<div algorithm>
+To <dfn>get the base network event data</dfn> given |request| and |is redirect|:
+
+1. Let |request data| be the result of [=get the request data=] with |request|.
+
+1. Let |navigation| be |request|'s [=request/navigation id=].
+
+1. Let |context id| be null.
+
+1. If |request|'s [=request/client=] is not null, let |related browsing
+   contexts| be the result of [=get related browsing contexts=] with
+   |request|'s [=request/client=]. Otherwise let |related browsing contexts| be
+   an empty set.
+
+1. If |request|'s [=request/window=] is an [=environment settings object=]:
+
+  1. Let |environment settings| be |request|'s [=request/window=]
+
+  1. If there is a [=/browsing context=] whose [=active window=] is |environment
+     settings|' [=environment settings object/global object=], let |context id|
+     be the [=browsing context id=] for that context.
+
+1. Let |timestamp| be a [=time value=] representing the current date and time in UTC.
+
+1. Let |params| be a [=map=] matching the <div>network.BaseParameters</div> production, with
+   the <code>request</code> field set to |request data|, the |navigation| field
+   set to <code>navigation</code>, the <code>context</code> field set to
+   |context id|, the <code>timestamp</code> field set to |timestamp|, and the
+   <code>isRedirect</code> field set to |is redirect|.
+
+1. Return (|related browsing contexts|, |params|)
+
+</div>
+
+#### The network.Cookie type #### {#type-network-Cookie}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl local-cddl">
+network.Cookie = {
+    name: text,
+    ? value: text,
+    ? binaryValue: [ uint ]
+    domain: text,
+    path: text,
+    ? expires: uint,
+    size: uint,
+    httpOnly: bool,
+    secure: bool,
+    sameSite: "strict" / "lax" / "none",
+};
+</pre>
+
+The <code>network.Cookie</code> type represents a cookie.
+
+If the cookie value can be represented as a UTF-8 encoded string, the
+<code>value</code> field will be present. Otherwise the <code>binaryValue</code>
+field will be present and consist of an array of integers representing the bytes
+of the cookie value.
+
+<div algorithm>
+To <dfn>get a cookie</dfn> given |stored cookie|:
+
+Note: The definitions of |stored cookie|'s fields are from [[COOKIES]], except
+samesite-flag, which is from [[SAME-SITE-COOKIES]].
+
+1. Let |name| be the result of [=UTF-8 decode=] with |stored cookie|'s name field.
```

Comment:
Double checked that this can't fail, because the default error mode is replacement.